core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -295,6 +295,13 @@ protected void validateNewDeleteFile(DeleteFile file) {
             formatVersion(),
             file.location());
         break;
+      case 4:
```

Comment:
why not combine the case 3 and 4  (fall through) as it is identical?

Reply:
this is mainly so that both versions are handled independently, even though right now they verify the same thing